File: graph/graph/src/main/groovy/net/twisterrob/gradle/graph/vis/d3/interop/TaskSerializer.java
package net.twisterrob.gradle.graph.vis.d3.interop;

import java.lang.reflect.Type;

import org.gradle.api.Task;

import com.google.gson.*;

public class TaskSerializer implements JsonSerializer<Task> {
	@Override public JsonElement serialize(Task task, Type type, JsonSerializationContext context) {
		return new JsonPrimitive(getKey(task));
	}

	public static String getKey(Task task) {
		return task.getPath();
	}
}

File: graph/src/main/java/net/twisterrob/gradle/graph/vis/d3/interop/TaskSerializer.kt
package net.twisterrob.gradle.graph.vis.d3.interop

import com.google.gson.JsonElement
import com.google.gson.JsonPrimitive
import com.google.gson.JsonSerializationContext
import com.google.gson.JsonSerializer
import org.gradle.api.Task
import java.lang.reflect.Type

class TaskSerializer : JsonSerializer<Task> {

	override fun serialize(task: Task, type: Type, context: JsonSerializationContext): JsonElement =
		JsonPrimitive(getKey(task))

	companion object {

		fun getKey(task: Task): String =
			task.path
	}
}
